### PR TITLE
Randomize upkeep check order in buckets

### PR DESCRIPTION
**Pull request by cmalec**
https://app.shortcut.com/chainlinklabs/story/32637/randomize-upkeep-check-order-in-buckets

### CODE REVIEW

**Comment by RyanRHall** [comment]
looks great - can we add a test? @cmalec

**Comment by RyanRHall** [comment]
Did you consider doing the shuffle at the DB level via `ORDER BY RANDOM`? I think this would be the preferred option if possible. The challenge is shuffling _and_ paginating...[see this SO](https://stackoverflow.com/questions/25482834/repeatable-pagination-with-randomly-ordered-rows).

**Comment by cmalec** [comment]
discussed shuffle performance vs adding sql complexity offline. Shuffle performance is not expensive even at 10k items it stays below a millisecond (avg ~179094 ns)

**Review by RyanRHall — APPROVED** [review]

File: core/services/keeper/orm_test.go [hunk 1]
```diff
@@ -135,6 +136,49 @@ func TestKeeperDB_BatchDeleteUpkeepsForJob(t *testing.T) {
...
+	assert.NoError(t, err)
+
+	require.Len(t, eligibleUpkeeps, 10)
+	ordered := [10]int64{}
+	shuffled := [10]int64{}
+	for i := 0; i < 10; i++ {
+		ordered[i] = upkeeps[i].UpkeepID
+		shuffled[i] = eligibleUpkeeps[i].UpkeepID
+	}
+	assert.NotEqualValues(t, ordered, shuffled)
```
Comment (RyanRHall) [inline core/services/keeper/orm_test.go:179]: 10 might be a kind of small number for this test. I bet with all our CI runs we will see it fail at least once. Can we decrease the probability of seeing a false positive? Or Make it more deterministic somehow?

**Review by RyanRHall — CHANGES_REQUESTED** [review]

File: core/services/keeper/orm_test.go (hunk 1 above)
Comment (cmalec) [reply core/services/keeper/orm_test.go:179]: sure ill up it to 100 so odds should be low

File: core/services/keeper/orm.go [hunk 2]
```diff
@@ -1,14 +1,18 @@
 package keeper
 
 import (
+	"math/rand"
+	"time"
+
 	"github.com/lib/pq"
 	"github.com/pkg/errors"
```
Comment (begmaroman) [inline core/services/keeper/orm.go:9]: Cosmetic: could you please remove this empty line

File: core/services/keeper/orm_test.go [hunk 3]
```diff
@@ -9,14 +9,15 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
```
Comment (begmaroman) [inline core/services/keeper/orm_test.go:11]: Cosmetic: could you please remove this empty line

**Review by begmaroman — DISMISSED** [review]
LGTM, just left minor comments

File: core/services/keeper/orm.go (hunk 2 above)
Comment (cmalec) [reply core/services/keeper/orm.go:9]: im using goimports sorting. is that not the format we use? what should i be using? I have my ide set to follow https://www.notion.so/chainlink/IDE-Settings-356dfa857ed54af99764b05a7a59082d

File: core/services/keeper/orm_test.go [hunk 4]
```diff
@@ -135,6 +135,139 @@ func TestKeeperDB_BatchDeleteUpkeepsForJob(t *testing.T) {
...
+		newUpkeep(registry, 90),
+		newUpkeep(registry, 91),
+		newUpkeep(registry, 92),
+		newUpkeep(registry, 93),
+		newUpkeep(registry, 94),
+		newUpkeep(registry, 95),
+		newUpkeep(registry, 96),
+		newUpkeep(registry, 97),
+		newUpkeep(registry, 98),
+		newUpkeep(registry, 99),
```
Comment (RyanRHall) [inline core/services/keeper/orm_test.go:248]: Can we just create these in a loop instead of hard coding 100 lines?

**Review by RyanRHall — CHANGES_REQUESTED** [review]

File: core/services/keeper/orm_test.go (hunk 4 above)
Comment (cmalec) [reply core/services/keeper/orm_test.go]: fixed

**Review by RyanRHall — DISMISSED** [review]

**Review by RyanRHall — APPROVED** [review]

**Pull request merged**